### DISCUSSION TITLE
Matplotlib now allows Vectorized hist style parameters

### DISCUSSION THREAD

**Discussion by EwoutH** [mwaskom/seaborn#3776] [Q&A] [answered]
Matplotlib now allows Vectorized hist style parameters, which might be useful for the Seaborn implementation of histplot.

- https://matplotlib.org/devdocs/users/next_whats_new/histogram_vectorized_parameters.html

The parameters hatch, edgecolor, facecolor, linewidth and linestyle of the [hist](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.hist.html#matplotlib.axes.Axes.hist) method are now vectorized. This means that you can pass in individual parameters for each histogram when the input x has multiple datasets.

```Python
import matplotlib.pyplot as plt
import numpy as np
np.random.seed(19680801)

fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2, figsize=(9, 9))

data1 = np.random.poisson(5, 1000)
data2 = np.random.poisson(7, 1000)
data3 = np.random.poisson(10, 1000)

labels = ["Data 1", "Data 2", "Data 3"]

ax1.hist([data1, data2, data3], bins=range(17), histtype="step", stacked=True,
         edgecolor=["red", "green", "blue"], linewidth=[1, 2, 3])
ax1.set_title("Different linewidths")
ax1.legend(labels)

ax2.hist([data1, data2, data3], bins=range(17), histtype="barstacked",
         hatch=["/", ".", "*"])
ax2.set_title("Different hatch patterns")
ax2.legend(labels)

ax3.hist([data1, data2, data3], bins=range(17), histtype="bar", fill=False,
         edgecolor=["red", "green", "blue"], linestyle=["--", "-.", ":"])
ax3.set_title("Different linestyles")
ax3.legend(labels)

ax4.hist([data1, data2, data3], bins=range(17), histtype="barstacked",
         facecolor=["red", "green", "blue"])
ax4.set_title("Different facecolors")
ax4.legend(labels)

plt.show()
```
![](https://matplotlib.org/devdocs/_downloads/6dc4fcd05a7af088beab36f014c1b768/histogram_vectorized_parameters-1.2x.png)

**Comment by mwaskom** [comment, marked as answer]
Neat — but `histplot` does not use `ax.hist` under the hood.

**Reply by EwoutH** [reply]
Thanks, didn’t know that
